src/Sarif.Viewer.VisualStudio/ErrorList/SarifAbsentResultsTableDataSource.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace Microsoft.Sarif.Viewer.ErrorList
+{
+    // A "stub" data source whose sole purpose is to cause the Category column
+    // to be displayed. It never actually adds entries to the Error List.
+    //
+    // For an explanation of why these this source is necessary, see the comment
+    // near the top of SarifResultTableEntry.cs
+    internal class SarifAbsentResultsTableDataSource : SarifTableDataSourceBase
+    {
+        private static SarifAbsentResultsTableDataSource _instance;
+
+        private SarifAbsentResultsTableDataSource()
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            this.Initialize(SarifResultTableEntry.AbsentResultColumns);
+        }
+
+        public static SarifAbsentResultsTableDataSource Instance =>
```

Comment:
>static [](start = 15, length = 6)

So sad, anyway to make this not static?

Reply:
Using MEF services local to the extension (not promoted) is the way to remove statics.

I could dig up lots of Watson hangs in VS due to the way the CLR initializes statics. (Let's just say it involves a global lock and it's super easy to deadlock between VS's UI thread and initialization of a static object)

---
In reply to: [538922382](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538922382) [](ancestors = 538922382)

Reply:
Ok, good advice. I will look at this tomorrow.

---
In reply to: [538923639](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538923639) [](ancestors = 538923639,538922382)